pandas/tests/io/test_gcs.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from pandas import DataFrame, read_csv
+from pandas.compat import BytesIO
+from pandas.io.common import is_gcs_url
+from pandas.util import _test_decorators as td
+
+
+def test_is_gcs_url():
+    assert is_gcs_url("gcs://pandas/somethingelse.com")
+    assert is_gcs_url("gs://pandas/somethingelse.com")
+    assert not is_gcs_url("s3://pandas/somethingelse.com")
+
+
+@td.skip_if_no('gcsfs')
+def test_read_csv_gcs():
+    try:
+        from unittest.mock import patch
+    except ImportError:
+        from mock import patch
+
+    with patch('gcsfs.GCSFileSystem') as MockFileSystem:
+        instance = MockFileSystem.return_value
+        instance.open.return_value = BytesIO(b'a,b\n1,2\n3,4')
+        df = read_csv('gs://test/test.csv')
+
+    assert isinstance(df, DataFrame)
+    assert len(df == 2)
+    assert all(df.columns == ['a', 'b'])
+
+
+@td.skip_if_no('gcsfs')
+def test_gcs_get_filepath_or_buffer():
+    try:
+        from unittest.mock import patch
+    except ImportError:
+        from mock import patch
+
+    with patch('pandas.io.gcs.get_filepath_or_buffer') as MockGetFilepath:
+        MockGetFilepath.return_value = (BytesIO(b'a,b\n1,2\n3,4'), None, None,
+                                        False)
+        df = read_csv('gs://test/test.csv')
+
+    assert isinstance(df, DataFrame)
```

Comment:
same

Reply:
can you add other dtypes such as float, datetime, string, and some NaNs

Reply:
Done